Deployment/Kubernetes/TensorRT-LLM_Autoscaling_and_Load_Balancing/chart/templates/deployment.yaml

```diff
@@ -0,0 +1,304 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- $hostRootPath := "/triton" }}
+{{- $image_name := "" }}
+{{- with $.Values.triton }}
+{{-   with .image }}
+{{-     $image_name = required "Property '.triton.image.name' is required." .name }}
+{{-   else }}
+{{-     fail "Property '.triton.image' is required." }}
+{{-   end }}
+{{- else }}
+{{-   fail "Property '.triton' is required" }}
+{{- end }}
+{{- $model_name := "" }}
+{{- $model_dt := "float16" }}
+{{- $model_pp := 1 }}
+{{- $model_tp := 1 }}
+{{- with $.Values.kubernetes }}
+{{-   with .hostRootPath }}
+{{-     $hostRootPath = . }}
+{{-   end }}
+{{- end }}
+{{- with $.Values.model }}
+{{-   $model_name = required "Property '.model.name' is required." .name }}
+{{-   with .tensorrtLlm }}
+{{-     with .dataType }}
+{{-       $model_dt = . }}
+{{-     end }}
+{{-     with .parallelism }}
+{{-       with .pipeline }}
+{{-         $model_pp = (int .) }}
+{{-       end }}
+{{-       with .tensor }}
+{{-         $model_tp = (int .) }}
+{{-       end }}
+{{-     end }}
+{{-   end }}
+{{- else }}
+{{-   fail "Property '.model' is required." }}
+{{- end }}
+{{- $model_lower := lower $model_name }}
+{{- $model_upper := upper $model_name }}
+{{- $model_gpus := mul $model_pp $model_tp }}
+{{- $triton_cpu := 4 }}
+{{- if lt $triton_cpu $model_gpus }}
+{{-   $triton_cpu = $model_gpus }}
+{{- end }}
+{{- $triton_memory := printf "%dGi" (mul $triton_cpu 8) }}
+{{- with $.Values.triton }}
+{{-   with .image }}
+{{-     with .name }}
+{{-       $image_name = . }}
+{{-     end }}
+{{-   end }}
+{{-   with .resources }}
+{{-     with .cpu }}
+{{-       $triton_cpu = (int .) }}
+{{-     end }}
+{{-     with .memory }}
+{{-       $triton_memory = . }}
+{{-     end }}
+{{-   end }}
+{{- end }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $.Release.Name }}
+  labels:
+    app: {{ $.Release.Name }}
+{{- with $.Values.kubernetes }}
+{{-   with .labels }}
+{{      toYaml . | indent 4 }}
+{{-   end }}
+{{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ $.Release.Name }}
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: {{ $.Release.Name }}
+        app.kubernetes.io/component: server
+{{- with $.Values.kubernetes }}
+{{-   with .labels }}
+{{      toYaml . | indent 8 }}
+{{-   end }}
+{{- end }}
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: nvidia.com/gpu
+                operator: Exists
```

Comment:
Because you didn't follow the instructions at the top of the document?

I get that the gpu-feature-discovery service will add the label recommended by you, but can you check that there aren't any failed pods, jobs, etc. related to nodes that lack a GPU?

Reply:
My point is basically just -- why ask the users to manually add the `nvidia.com/gpu` label when they automatically get the `nvidia.com/gpu.present` label from gpu-feature-discovery, and if they don't have gpu-feature-discovery they can manually add `nvidia.com/gpu.present` just as easily